File: app/src/main/java/com/dvor/my/mydvor/service/ServiceAdapter.java
package com.dvor.my.mydvor.service;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.dvor.my.mydvor.R;
import com.dvor.my.mydvor.data.Service;

import java.util.List;

public class ServiceAdapter extends ArrayAdapter<Service> {

    private LayoutInflater inflater;
    private int layout;
    private List<Service> services;
    private Context context;

    public ServiceAdapter(Context context, int resource, List<Service> services) {
        super(context, resource, services);
        this.services = services;
        this.layout = resource;
        this.inflater = LayoutInflater.from(context);
        this.context=context;

    }
    public View getView(int position, View convertView, ViewGroup parent) {

        View view=inflater.inflate(this.layout, parent, false);

        ImageView imgView = view.findViewById(R.id.image);
        TextView titleView = view.findViewById(R.id.title);
        TextView textView = view.findViewById(R.id.text);
        final TextView phone = view.findViewById(R.id.phone);

        Service service = services.get(position);

       // imgView.setImageResource(service.getImgResource());
        titleView.setText(service.getTitle());
        textView.setText(service.getText());
        phone.setText(service.getPhone());

        phone.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent intent = new Intent(Intent.ACTION_DIAL, Uri.fromParts("tel", phone.getText().toString(), null));
                context.startActivity(intent);

            }
        });

        return view;
    }
}

File: app/src/main/java/com/dvor/my/mydvor/service/ServiceAdapter.kt
package com.dvor.my.mydvor.service

import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import com.dvor.my.mydvor.R
import com.dvor.my.mydvor.data.Service

class ServiceAdapter(private val context: FragmentActivity?, private val layout: Int, private val services: List<Service>) : ArrayAdapter<Service>(context!!, layout, services) {

    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val view = inflater.inflate(this.layout, parent, false)

        val imgView = view.findViewById<ImageView>(R.id.image)
        val titleView = view.findViewById<TextView>(R.id.title)
        val textView = view.findViewById<TextView>(R.id.text)
        val phone = view.findViewById<TextView>(R.id.phone)

        val service = services[position]

        // imgView.setImageResource(service.getImgResource());
        titleView.text = service.title
        textView.text = service.text
        phone.text = service.phone

        phone.setOnClickListener {
            val intent = Intent(Intent.ACTION_DIAL, Uri.fromParts("tel", phone.text.toString(), null))
            context?.startActivity(intent)
        }

        return view
    }
}